### PR TITLE
Add an attribute to pkg_tar to remap paths in the tar files.

### PR DESCRIPTION
**Pull request by katre**

### CODE REVIEW

**Comment by katre** [comment]
It seems that #6601 is just yaks all the way down.

File: tools/build_defs/pkg/pkg.bzl [hunk 1]
```diff
@@ -200,6 +211,7 @@ _real_pkg_tar = rule(
         "empty_files": attr.string_list(),
         "include_runfiles": attr.bool(default = False, mandatory = False),
         "empty_dirs": attr.string_list(),
+        "remap_paths": attr.string_dict(),
```
Comment (aiuto) [inline tools/build_defs/pkg/pkg.bzl]: can you add to README.md.
Even better if we move the docs to the attrs and generate with Stardoc, but that should be a different CL.

Comment (katre) [reply tools/build_defs/pkg/pkg.bzl]: Done!

**Comment by alex-xnor** [comment]
Is there a reason that `_remap` is using `replace` here? It seems like it might lead to erroneous results. For example, given the path `pea/chickpea/legume` and the mapping `{"pea/": "foods/"}`, it seems like it would result in `foods/chickfoods/legume` rather than `foods/chickpea/legume`. Is this intentional?

**Comment by katre** [comment]
No, that is not intentional, that's bad coding on my part. I'll send a PR to fix that ASAP.

**Comment by katre** [comment]
Filed #6778, and sent #6779 to fix.
